src/Framework/PaymentGateways/Commands/SubscriptionComplete.php

```diff
@@ -16,13 +22,32 @@ class SubscriptionComplete implements GatewayCommand {
      * @var string
      */
     public $gatewaySubscriptionId;
+    /**
+     * The subscription status.
+     *
+     * @var SubscriptionStatus
+     */
+    public $subscriptionStatus;
+    /**
+     * The donation status.
+     *
+     * @var DonationStatus
+     */
+    public $donationStatus;
 
     /**
+     * @unreleased Add support for donation and subscription status.
      * @since 2.18.0
      */
-    public function __construct(string $gatewayTransactionId, string $gatewaySubscriptionId)
-    {
+    public function __construct(
+        $gatewayTransactionId,
+        string $gatewaySubscriptionId,
+        SubscriptionStatus $subscriptionStatus = null,
+        DonationStatus $donationStatus = null
```

Comment:
@ravinderk I'm starting to second-guess what we're doing, here, especially since it's a public API. I'm tempted to make a `SubscriptionProcessing` command that has different defaults and is more flexible. It may be worth distinguishing between "the subscription is created and everything is complete" versus " the subscription is created but still being processed in the gateway".

Mind weighing in on this, @jonwaldstein?

Reply:
@JasonTheAdams I am all in for the `SubscriptionProcessing` gateway command.

Reply:
@JasonTheAdams Agreed, a subscription should not have to worry about their status when returning a command.  The command should imply their status and intentions.  If a subscription is still processing then they should not be returning a complete command.

Reply:
@ravinderk Yeah, let's do that then, please! Let's put `SubscriptionComplete` back the way it was.

Reply:
resolved